src/main/java/io/cdap/plugin/sap/transformer/ODataEntryToRecordTransformer.java

```diff
@@ -0,0 +1,419 @@
+      .set(SapODataConstants.GEO_COLLECTION_POLYGONS_FIELD_NAME, polygons)
+      .set(SapODataConstants.GEO_COLLECTION_MULTI_POINTS_FIELD_NAME, multiPoints)
+      .set(SapODataConstants.GEO_COLLECTION_MULTI_LINE_STRINGS_FIELD_NAME, multiLineStrings)
+      .set(SapODataConstants.GEO_COLLECTION_MULTI_POLYGONS_FIELD_NAME, multiPolygons)
+      .build();
+  }
+
+  private StructuredRecord geospatialRecordOf(String typeName, List coordinates, Schema schema) {
+    return StructuredRecord.builder(schema)
+      .set(SapODataConstants.GEOSPATIAL_TYPE_FIELD_NAME, typeName)
+      .set(SapODataConstants.GEOSPATIAL_COORDINATES_FIELD_NAME, coordinates)
+      .build();
+  }
+
+  private List<Double> extractCoordinates(Point point) {
+    return Arrays.asList(point.getX(), point.getY());
+  }
+
+  private List<List<Double>> extractCoordinates(LineString lineString) {
+    List<List<Double>> coordinates = new ArrayList<>();
+    Iterator<Point> pointIterator = lineString.iterator();
+    if (pointIterator == null) {
+      return coordinates;
+    }
+    pointIterator.forEachRemaining(p -> coordinates.add(extractCoordinates(p)));
+
+    return coordinates;
+  }
+
+  private List<List<List<Double>>> extractCoordinates(Polygon polygon) {
+    List<List<List<Double>>> coordinates = new ArrayList<>();
+    if (polygon.getExterior() != null && polygon.getExterior().iterator() != null) {
+      List<List<Double>> exteriorCoordinates = new ArrayList<>();
+      polygon.getExterior().iterator().forEachRemaining(p -> exteriorCoordinates.add(extractCoordinates(p)));
+      coordinates.add(exteriorCoordinates);
+    }
+
+    for (int i = 0; i < polygon.getNumberOfInteriorRings(); i++) {
+      Iterator<Point> interiorIterator = polygon.getInterior(i).iterator();
+      if (interiorIterator == null) {
+        continue;
+      }
+      List<List<Double>> interiorCoordinates = new ArrayList<>();
+      interiorIterator.forEachRemaining(p -> interiorCoordinates.add(extractCoordinates(p)));
+      coordinates.add(interiorCoordinates);
+    }
+
+    return coordinates;
+  }
+
+  private List<List<Double>> extractCoordinates(MultiPoint multiPoint) {
+    List<List<Double>> coordinates = new ArrayList<>();
+    Iterator<Point> pointIterator = multiPoint.iterator();
+    if (pointIterator == null) {
+      return coordinates;
+    }
+    pointIterator.forEachRemaining(p -> coordinates.add(extractCoordinates(p)));
+
+    return coordinates;
+  }
+
+  private List<List<List<Double>>> extractCoordinates(MultiLineString multiLineString) {
+    List<List<List<Double>>> coordinates = new ArrayList<>();
+    Iterator<LineString> lineStringIterator = multiLineString.iterator();
+    if (lineStringIterator == null) {
+      return coordinates;
+    }
+    lineStringIterator.forEachRemaining(ls -> coordinates.add(extractCoordinates(ls)));
+
+    return coordinates;
+  }
+
+  private List<List<List<List<Double>>>> extractCoordinates(MultiPolygon multiPolygon) {
```

Comment:
This looks unusable. I don't know how anybody would use a list of lists of lists of lists. We should represent these objects similar to how they are represented as java objects.

For example, a Point is a record with 2 double fields - x and y.

A MultiPoint is a record with a single field 'coordinates' that is an array of Point records.

A Polygon is a record with multiple fields:
 an exterior field, which is an array of MultiPoint records, 
 an interior fields, which is an array of MultiPoint records,
 a numberOfInteriorRings field of type int

etc.

Reply:
Thanks. Removed from this PR. I will introduce this changed in the next PR.